app/account/page.tsx

```diff
@@ -35,6 +35,15 @@ export default function AccountPage() {
     throw new Error(error.message)
   }
 
+  const userInfoQuery = JSON.stringify({
+    name: user?.name,
+    email: user?.email,
+    profile_picture_url: user?.profile_picture_url ?? null
+  })
+
+  // Generate the URL with the query parameter
+  const href = `/account/edit?userInfo=${encodeURIComponent(userInfoQuery)}`
+
```

Comment:
Here, instead of making another getUserQuery, I passed user information as a prop and retrieve the data by using useSearchParams in the account edit page. I'm not entirely sure if it's the best approach, especially since the route becomes longer due to including the name, email, and profile image URL, like this:
`http://localhost:3000/account/edit?userInfo=%7B%22name%22%3A%22Sachi%20Goto%20Name%20changed!%22%2C%22email%22%3A%22test123%40gmail.com%22%2C%22profile_picture_url%22%3Anull%7D`

Reply:
As long as we use the query of Apollo client, in this case, `getUserQuery,` it has to be cashed already when the user sees the account details page. So we don't need to pass the information using a URL query.

Check the documentation!
https://www.apollographql.com/docs/react/why-apollo

Reply:
oh that's really cool! Good to know!